src/mds/Server.cc

```diff
@@ -578,6 +578,39 @@ void Server::handle_client_session(const cref_t<MClientSession> &m)
   if (logger)
     logger->inc(l_mdss_handle_client_session);
 
+  auto& addr = session->info.inst.addr;
+  session->set_client_metadata(client_metadata_t(m->metadata, m->supported_features, m->metric_spec));
```

Comment:
This will set the metadata for all the cases, before it will set the metadata only when opening the session.